### ISSUE TITLE
Implicit module name duplication

### ISSUE THREAD

**Issue by Emoun** [Emoun/duplicate#7]
Since [procedural macros will soon support hygiene](https://github.com/rust-lang/rust/pull/68717#issuecomment-623197503), it would be appropriate to implement implicit module name duplication.

if `duplicate` is applied to a module, the user is currently required to make sure that the module has a unique name for every duplicate. This is tedious and must be done every time. To improve the user experience, `duplicate` could use hygiene to allow all duplicates to keep the same module name, but with different hygiene. This means the user no longer has to manually assign names to each module duplicate.

The rule could be that if `duplicate` is applied to a module directly, hygienic renaming will kick in, however, if the module is already renamed by a substitution identifier, the hygienic renaming won't happen. This should therefore be a non-breaking change.

**Comment by Emoun** [comment]
If we are doing to enable easy module duplicate, maybe also include a `use super::*` in the module?

**Comment by Emoun** [comment]
> 
> 
> If we are doing to enable easy module duplicate, maybe also include a `use super::*` in the module?

No, as the user should be allowed to decide for themselves whether the outer scopes imports should be available for each module. This will therefore force a specific usage pattern on users.

**Comment by Emoun** [comment]
Using hygiene to reuse the module name may not be appropriate everywhere. F.eks. if tests are duplicated this way, there is no way to know which module failed, as they would all have the same name (to the user.)
It might therefore be useful to create some way for the user to easily change the name. E.g. maybe the first substitution identifier's substitution could be appended to the module name automatically. F.eks:

```
#[duplicate( 
  directedness; 
  [Directed]; 
  [Undirected] )
]
mod module {
....
}
```
Could expand to:
```
mod module_directed {
....
}
mod module_undirected {
....
}
```
Here notice that the `directedness` identifier is used inside the module. The rule could be that if the first identifier's substitutions are pure identifiers, then they are appended to the module's name in lower case.

**Comment by Emoun** [comment]
Implemented and released in v0.2.6.
